pyo3-macros-backend/src/pymethod.rs

```diff
@@ -491,6 +491,12 @@ fn impl_arg_param(
         (None, false) => quote! { panic!("Failed to extract required method argument") },
     };
 
+    let extract = if let Some(expr) = &arg.attrs.from_py_with {
+        quote! {#expr(_obj.extract()?).map_err(#transform_error)?}
```

Comment:
This looks like it'll allow the `from_py_with` function to accept any type which is convertible from Python, instead of just `PyAny`.

This is probably more flexible than we need it, so I wonder if we should dial it back a bit and force the functions to take `PyAny`?

```suggestion
        quote! {#expr(_obj).map_err(#transform_error)?}
```

Also as written if the `.extract()` call failed it would not be passed through `#transform_error`, so if we want to keep support for typed inputs to the extraction function we should have:

```suggestion
        quote! {_obj.extract().and_then(#expr).map_err(#transform_error)?}
```

Reply:
I think you're right, we should restrict inputs to `&PyAny`.
 
One can downcast or extract to a specific type explicitly in the conversion function, if neccessary. 
Also, seeing a non-python type in a conversion function signature might be confusing.

Reply:
Sounds good to me! If someone has a really compelling reason why something other than `&PyAny` should be supported later, we can revisit then.